docs/DeveloperGuide.md

```diff
@@ -346,42 +347,420 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
+|[Search](#use-case-search)|Displays a list of tasks with matching keywords as queried|
+|[View Overdue Tasks](#use-case-view-overdue-tasks)|Displays a list of tasks that are overdue|
+|[Undo Previous Command](#use-case-undo-previous-command)|Undo the previous command|
+|[Repeat Task](#use-case-repeat-task)|Repeat the task every week|
+|[Add Tags to Created Task](#use-case-add-tags-to-created-task)|Add one or more tags to a task to label it|
+|[Customise Message of the Day](#use-case-customise-message-of-the-day)|Add a message of the day that is displayed each time the program starts up|
+|[Scroll Through Previous Commands](#use-case-scroll-through-previous-commands)|Iterate through previously executed commands|
 
-#### Use case: Delete person
+---
 
-**MSS**
+### Use Case: Display Help
 
-1. User requests to list persons
-2. AddressBook shows a list of persons
-3. User requests to delete a specific person in the list
-4. AddressBook deletes the person <br>
-Use case ends.
+#### Main Success Scenario:
 
-**Extensions**
+1. User requests to display help.
+2. System displays the help messages.
+3. Use case ends.
 
-2a. The list is empty
+##### Extensions
 
-> Use case ends
+1a. User entered an invalid command.
 
-3a. The given index is invalid
+> 1a1. System displays help messages.
+> 1a2. Use case ends.
 
-> 3a1. AddressBook shows an error message <br>
-  Use case resumes at step 2
+---
 
-{More to be added}
+### Use Case: Add Task
+
+#### Main Success Scenario:
+
+1. User requests to add a task.
+2. System accepts the task.
+3. Use case ends.
+
+##### Extensions
+
+1a. User entered an invalid command.
+
+> 1a1. System display unsuccessful message.
+> 1a2. Use case ends.
+
+---
+
+### Use case: Delete Task
+#### Main Success Scenario:
+1. User requests to list tasks.
+2. System displays tasks.
+3. User deletes a task by its index.
+4. System displays new list without the deleted task.
+5. Use case ends.
+
+##### Extensions
+2a. There is no list.
+> Use case ends.
+
+3a. The given index is invalid.
+> 3a1. System gives an error message.
+Use case resumes at step 2.
+
+---
+
+### Use Case: Complete Task
+
+#### Main Success Scenario:
+1. User requests to list tasks.
+2. System displays tasks.
+3. User selects a task based on its task number and marks it as completed.
+4. System marks the task as completed.
+5. Use case ends.
+
+#### Extension
+2a. There is no list.
+> Use case ends.
+
+3a. The task has already been completed.
+> 3a1. System shows error message that the task has already been completed.
+> 3a2. Use case ends.
+
+3b. The provided task number does not exist.
+> 3b1. System shows error message that the task number does not exist.
+> 3b2. Use case ends.
+
+---
+
+### Use case: Edit Task
+#### Main Success Scenario:
+1. User requests to list uncompleted tasks.
+2. System displays uncompleted tasks.
+3. User edits a task.
```

Comment:
both the user guide and this use case doesn't fully describe how edit works. There is no where that the user can enter new details to the task that is being edited